Dockerfile

```diff
@@ -1,10 +1,27 @@
-FROM ruby:3.2.1-alpine3.17
+FROM ruby:3.2.1-alpine3.17 as base
 
-ARG SUPERCRONIC_PLATFORM=amd64
-ARG SUPERCRONIC_SHA1SUM=6817299e04457e5d6ec4809c72ee13a43e95ba41
+FROM base AS base-amd64
+ENV SUPERCRONIC_SHA1SUM=6817299e04457e5d6ec4809c72ee13a43e95ba41
 
-ENV SUPERCRONIC_URL=https://github.com/aptible/supercronic/releases/download/v0.2.24/supercronic-linux-${SUPERCRONIC_PLATFORM} \
-    SUPERCRONIC=supercronic-linux-${SUPERCRONIC_PLATFORM}
+FROM base AS base-arm64
+ENV SUPERCRONIC_SHA1SUM=fce407a3d7d144120e97cfc0420f16a18f4637d9
+# https://github.com/sparklemotion/nokogiri/issues/2414
+# Required for ARM (otherwise nokogiri breaks when viewing network graph)
+RUN apk add --update --no-cache gcompat
+
+FROM base AS base-arm
+ENV SUPERCRONIC_SHA1SUM=fad9380ed30b9eae61a5b1089f93bd7ee8eb1a9c
+RUN apk add --update --no-cache gcompat
+
+ARG TARGETARCH
+FROM base-$TARGETARCH AS pb-dev
+
+RUN env
```

Comment:
Still want this line or was this just for debugging?

Reply:
I think its required to source the `TARGET_ARCH` value used in the `ARG` value below but will check without it